docs/getting-started/provider-gcp.md

```diff
@@ -0,0 +1,227 @@
+
+{{< hint type="note" >}}
+Other authentication methods exist and are beyond the scope of this guide. The [Provider documentation](https://marketplace.upbound.io/providers/upbound/provider-gcp/latest/docs/configuration) contains information on alternative authentication methods. 
+{{< /hint >}}
+
+#### Generate a GCP service account JSON file
+For basic user authentication, use a Google Cloud service account JSON file. 
+
+{{< hint type="tip" >}}
+The [GCP documentation](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) provides information on how to generate a service account JSON file.
+{{< /hint >}}
+
+Save this JSON file as `gcp-credentials.json`
+
+{{< hint type="note" >}}
+The [Configuration](https://marketplace.upbound.io/providers/upbound/provider-gcp/latest/docs/configuration) section of the Provider documentation describes other authentication methods.
+{{< /hint >}}
+
+#### Create a Kubernetes secret with the GCP credentials
+<!-- vale gitlab.Substitutions = NO -->
+<!-- ignore .json file name -->
+A Kubernetes generic secret has a name and contents. Use {{< hover label="kube-create-secret" line="1">}}kubectl create secret{{< /hover >}} to generate the secret object named {{< hover label="kube-create-secret" line="2">}}gcp-secret{{< /hover >}} in the {{< hover label="kube-create-secret" line="3">}}crossplane-system{{</ hover >}} namespace.  
+Use the {{< hover label="kube-create-secret" line="4">}}--from-file={{</hover>}} argument to set the value to the contents of the  {{< hover label="kube-create-secret" line="4">}}gcp-credentials.json{{< /hover >}} file.
+<!-- vale gitlab.Substitutions = YES -->
+
+```shell {label="kube-create-secret",copy-lines="all"}
+kubectl create secret \
+generic gcp-secret \
+-n crossplane-system \
+--from-file=creds=./gcp-credentials.json
+```
+
+View the secret with `kubectl describe secret`
+
+{{< hint type="note" >}}
+The size may be larger if there are extra blank spaces in your text file.
+{{< /hint >}}
+
+```shell
+kubectl describe secret gcp-secret -n crossplane-system
+Name:         gcp-secret
+Namespace:    crossplane-system
+Labels:       <none>
+Annotations:  <none>
+
+Type:  Opaque
+
+Data
+====
+creds:  2330 bytes
+```
+
+### Create a ProviderConfig
+A `ProviderConfig` customizes the settings of the GCP Provider.  
+
+Apply the {{< hover label="providerconfig" line="2">}}ProviderConfig{{</ hover >}}. Include your {{< hover label="providerconfig" line="7" >}}GCP project ID{{< /hover >}}.
+
+{{< hint type="warning" >}}
+Add your GCP `project ID` into the output below. 
+{{< /hint >}}
+
+```yaml {label="providerconfig",copy-lines="all"}
+cat <<EOF | kubectl apply -f -
+apiVersion: gcp.upbound.io/v1beta1
+kind: ProviderConfig
+metadata:
+  name: default
+spec:
+  projectID: <PROJECT_ID>
+  credentials:
+    source: Secret
+    secretRef:
+      namespace: crossplane-system
+      name: gcp-secret
+      key: creds
+EOF
```

Comment:
This isn't a YAML document, it's a snippet of shell.

Reply:
Same as before, the ```` ```yaml ```` is a hint to the highlighter. Using YAML highlighting here is much more readable than shell. 

The hint used is not exposed in the docs.

Reply:
Consider putting a (hidden) comment to explain your thinking here, to help future maintainers.

Reply:
Reasoning has been added to the [contributing guide](https://github.com/crossplane/crossplane/pull/3396/files#diff-93426d953f03b733adbbb574ffd2717d0892773a464c0d301d1b6c9095d67b15R191).

Resolving this conversation (unresolve if you're not satisfied with this)